### PR TITLE
Highlight Component

### PR DESCRIPTION
**Pull request by Ahmet-K**
Adding `Highlight` component so that user can easily add a highlight.

Tasks:
- [x] Add unit tests
- [x] Add Application code with commented props
- [x] Add the ability to specify other props (...props)
- [x] Add Demo use of component
- [x] Add to storybook documentation file with examples (.mdx)  in a section called Typography
- [x] Add to live demo

https://github.com/Capgemini/dcx-react-library/issues/498

Screenshot 2023-06-06 at 16 04 12

### CODE REVIEW

File: src/highlight/Highlight.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  /**
+   * optional CSS class name
+   */
+  className?: string;
+  /**
+   * content of the highlight
+   */
+
+  children: JSX.Element | string;
+    // text: string | number;
```
Comment (daniele-zurico) [inline src/highlight/Highlight.tsx:14]: remove this comment please and align the above comment with this code

File: src/highlight/Highlight.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  /**
+   * Additional props/attributes
+   */
+  props?: React.HTMLAttributes<HTMLQuoteElement>;
```
Comment (daniele-zurico) [inline src/highlight/Highlight.tsx:19]: there's anything better than `HTMLQuoteElement`?

File: stories/Highlight/ClassBased.stories.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  render: function(args) {
+    return (
+      <Highlight {...args}>
+        This is the content of the highlighted text.{' '}
+      </Highlight>
+    );
+  },
+  args: {
+    className: 'highlight'
+  },
```
Comment (daniele-zurico) [inline stories/Highlight/ClassBased.stories.js:27]:
```suggestion
export const Basic = {
  name: 'Basic',
  args: {
    className: 'highlight',
    children: 'This is the content of the highlighted text.'
  },
```
Should work?

Comment (daniele-zurico) [reply stories/Highlight/ClassBased.stories.js:27]: I also don't see any class defined for `hightlight` ... you need to create a css file in this folder define some properties and include it here

File: src/highlight/Highlight.tsx (hunk 1 above)
Comment (Ahmet-K) [reply src/highlight/Highlight.tsx]: Removed.

File: src/highlight/Highlight.tsx (hunk 2 above)
Comment (Ahmet-K) [reply src/highlight/Highlight.tsx]: Replaced with HTMLElement.

File: stories/Highlight/ClassBased.stories.js (hunk 3 above)
Comment (Ahmet-K) [reply stories/Highlight/ClassBased.stories.js]: Done.

**Review by daniele-zurico — APPROVED** [review]

**Pull request merged**